modules/tensor_mechanics/doc/content/documentation/systems/UserObjects/GeneralizedPlaneStrainUserObject.md

```diff
@@ -1,19 +1,30 @@
-<!-- MOOSE Documentation Stub: Remove this when content is added. -->
-
 # GeneralizedPlaneStrainUserObject
 
-!alert construction title=Undocumented Class
-The GeneralizedPlaneStrainUserObject has not been documented, if you would like to contribute to MOOSE by
-writing documentation, please see [/generate.md]. The content contained on this page explains
-the typical documentation associated with a MooseObject; however, what is contained is ultimately
-determined by what is necessary to make the documentation clear for users.
-
 !syntax description /UserObjects/GeneralizedPlaneStrainUserObject
 
+The `GeneralizedPlaneStrainUserObject` calculates the values of residual and diagonal jacobian components for each given scalar out-of-plane strain variable. This object is usually set up by the [GeneralizedPlaneStrainAction](/GeneralizedPlaneStrainAction.md).
+
+For a given scalar out-of-plane strain variable, the equilibrium condition condition in the out-of-plane direction (e.g. $z$-direction) is given as
+\begin{equation}
+	\int_{A}{\sigma_{zz}dA} = \bar{N}_{zz}
+\end{equation}
+
+Thus, the residual corresponding to the scalar out-of-plane strain variable is
+
+\begin{equation}
+	R = \int_{A}{\sigma_{zz}dA} - \bar{N}_{zz}
+\end{equation}
+
+and the corresponding diagonal jacobian is
+
+\begin{equation}
+	K_{zz} = \frac{\partial R_{zz}}{\partial \epsilon_{zz}} = \int_{A}{\frac{\partial \sigma_{zz}}{\partial \epsilon_{zz}}dA} = \int_{A}{C_{2222}dA}
+\end{equation}
```

Comment:
My preference is to just have links from the other pages, rather than using !include, because !include gives us lots of opportunities to mess things up when we don't know that other pages are including the stuff we are editing.